doc/types.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package doc
+
+// ID represents a document's natural identifier. It's treated as immutable
+// by the indexing subsystem.
+type ID []byte
+
+// FieldValueType represents the different types of field values supported.
+type FieldValueType byte
+
+const (
+	// UnknownValueType demarks an unknown value type.
+	UnknownValueType FieldValueType = iota
+
+	// StringValueType demarks string value types.
+	StringValueType
+)
+
+// Term represents the value of a field.
+type Term []byte
```

Comment:
Was thinking about this some more, given that `Term` is often used to expressed a field and its value, maybe just `Value` would be more appropriate?

Reply:
Sure

Reply:
done